lib/evaluate/tools.h

```diff
@@ -72,9 +72,16 @@ template<typename A> bool IsVariable(const FunctionRef<A> &funcRef) {
     return false;
   }
 }
-template<typename A> bool IsVariable(const Expr<A> &expr) {
+template<typename T> bool IsVariable(const Expr<T> &expr) {
   return std::visit([](const auto &x) { return IsVariable(x); }, expr.u);
 }
+template<typename A> bool IsVariable(const std::optional<A> &x) {
```

Comment:
Is there a convention you are following as to when to use `T` and when to use `A`?

Reply:
I use `A` when it can be any C++ type in the compiler, and `T` when it can be any Fortran type in the program.  So `optional<A>` but `Expr<T>`.  I'm probably not entirely consistent with these.

Reply:
I think that I use `A` because `a` is idiomatic for an arbitrary type variable in Haskell.